sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -195,6 +197,11 @@ class SparkConnectPlanner(
           transformWithWatermark(rel.getWithWatermark)
         case proto.Relation.RelTypeCase.CACHED_LOCAL_RELATION =>
           transformCachedLocalRelation(rel.getCachedLocalRelation)
+        case proto.Relation.RelTypeCase.CHUNKED_CACHED_LOCAL_RELATION
```

Comment:
This should probably be more subtle. The parallelism is effectively set by the client. In some cases and with in-proper settings that can lead to poorer performance. So we may want to feature flag this. We may also want to add a threshold for this, that would make smaller relations leverage the old code path.